owltorrent-network/src/main/java/edu/rice/owltorrent/network/SocketConnector.java

```diff
@@ -0,0 +1,136 @@
+package edu.rice.owltorrent.network;
+
+import edu.rice.owltorrent.common.adapters.NetworkToStorageAdapter;
+import edu.rice.owltorrent.common.entity.Peer;
+import edu.rice.owltorrent.network.messagereader.BusyWaitMessageReader;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.net.Socket;
+import java.nio.ByteBuffer;
+import java.nio.channels.Channels;
+import java.nio.channels.ReadableByteChannel;
+import lombok.extern.log4j.Log4j2;
+
+/**
+ * Socket implementation of PeerConnector class.
+ *
+ * @author Lorraine Lyu, Max Yu
+ */
+@Log4j2(topic = "general")
+public class SocketConnector extends PeerConnector {
+  private final Socket peerSocket;
+  private DataOutputStream out;
+  private DataInputStream in;
+
+  private final Runnable listenForInput =
+      new Runnable() {
+        @Override
+        public void run() {
+          ReadableByteChannel channel = Channels.newChannel(in);
+          while (true) {
```

Comment:
Should not busy wait here, add a thread.sleep as a passed in config param

Reply:
So this would not really busy wait. The readMessage would block and wait for read instead.

Reply:
My bad